desktop/cmp/panel/PanelModel.ts

```diff
@@ -31,82 +102,59 @@ export class PanelModel extends HoistModel {
     //-----------------------
     // Immutable Properties
     //-----------------------
-    resizable;
-    resizeWhileDragging;
-    collapsible;
-    defaultSize;
-    minSize;
-    maxSize;
-    defaultCollapsed;
-    side;
-    renderMode;
-    refreshMode;
-    showSplitter;
-    showSplitterCollapseButton;
-    showHeaderCollapseButton;
-    showModalToggleButton;
-
-    @managed refreshContextModel;
-    @managed provider;
+    readonly resizable: boolean;
+    resizeWhileDragging: boolean;
+    readonly collapsible: boolean;
+    readonly defaultSize: number;
+    readonly minSize: number;
+    readonly maxSize: number;
+    readonly defaultCollapsed: boolean;
+    readonly side: 'top'|'bottom'|'left'|'right';
+    readonly renderMode: RenderMode;
+    readonly refreshMode: RefreshMode;
+    readonly showSplitter: boolean;
+    readonly showSplitterCollapseButton: boolean;
+    readonly showHeaderCollapseButton: boolean;
+    readonly showModalToggleButton: boolean;
+
+    @managed modalSupportModel: ModalSupportModel;
+    @managed refreshContextModel: RefreshContextModel;
+    @managed provider: PersistenceProvider;
 
 
     //---------------------
     // Observable State
     //---------------------
     /** Is the Panel rendering in a collapsed state? */
-    @observable collapsed = false;
+    @observable
+    collapsed: boolean = false;
 
     /** Size in pixels along sizing dimension.  Used when object is *not* collapsed. */
-    @observable size = null;
+    @observable
+    size: number = null;
 
     /** Is this panel currently resizing? */
-    @observable isResizing = false;
+    @observable
+    isResizing: boolean = false;
 
     /** Is the panel rendering in its modal view state? Observable property. */
-    get isModal() {
+    get isModal(): boolean {
         return !!this.modalSupportModel?.isModal;
     }
 
-    get hasModalSupport() {
+    get hasModalSupport(): boolean {
         return !!this.modalSupportModel;
     }
 
-    get isActive() {
+    get isActive(): boolean {
         return !this.collapsed;
     }
 
     //-----------------
     // Implementation
     //-----------------
     _resizeRef;
```

Comment:
Should we lose the underscore and mark this as private?

Reply:
can't mark as private -- used in implementation classes -- leaving for now....